src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -347,6 +358,115 @@ def _retrieve_pre_processed_docs(tool: str,
+
+    for date_format in formats_to_try:
+        try:
+            parsed_date = datetime.strptime(date_string, date_format)
+            iso8601_date = parsed_date.date().isoformat()
+            return iso8601_date
+        except ValueError:
+            continue
+
+    raise ValueError("Unrecognized date format")
+
+
+def _convert_values_to_type(docs: list[dict], key: str, col_type: ColumnType):
+    # Convert the values of a column to the specified type
+    values = []
+    for doc in docs:
+        try:
+            value = doc[key]
+            value = None if value in NONE_STR else value
+            if value is None:
+                doc[key] = value
+                continue
+
+            if col_type == ColumnType.INT:
+                doc[key] = int(value)
+            elif col_type == ColumnType.FLOAT:
+                doc[key] = float(value)
+            elif col_type == ColumnType.STRING:
+                doc[key] = str(value)
+            elif col_type == ColumnType.DATE:
+                doc[key] = _convert_to_iso8601(value)
+            elif col_type == ColumnType.ENUM:
+                pass
+            else:
+                raise ValueError(f'casting not implemented for {col_type}')
+
+            values.append(doc[key])
+        except KeyError as e:
+            raise ValueError(f'Unable to find key: {key} in {doc}') from e
+        except ValueError as e:
+            raise ValueError(f'Unable to convert value: {key} from {doc} to type: {col_type}') from e
+
+    return values
+
+
+def process_columnar_meta(
+        docs: list[dict],
+        kbase_collection: str,
+        load_ver: str,
+        allow_missing_metadata: bool = True,
+):
+    """
+    Process the columnar metadata for the genome attributes.
+
+    :param docs: the list of documents
+    :param kbase_collection: the KBase collection name
+    :param load_ver: the load version
+    :param allow_missing_metadata: whether to allow missing metadata
+    """
+
+    spec = load_spec(names.GENOME_ATTRIBS_PRODUCT_ID, kbase_collection)
+    columns = list()
+    for col_spec in spec.columns:
+        key, col_type = col_spec.key, col_spec.type
+        if key not in docs[0]:
+            error_message = f'Unable to find column: {key} in the docs'
+            if allow_missing_metadata:
+                # some genome attribute tools may not have been executed for specific collection
+                print(error_message)
+                continue
+            else:
+                raise ValueError(error_message)
+
+        values = _convert_values_to_type(docs, key, col_type)
+        min_value, max_value, enum_values = None, None, None
+        if col_type in [ColumnType.INT, ColumnType.FLOAT, ColumnType.DATE]:
+            if values:
+                min_value, max_value = min(values), max(values)
+            else:
+                # AttributesColumn validates min_value and max_value, so we need to set them to 0 if column is empty
```

Comment:
It happens to one of my mini test collection. But I think in general it won't happen. I just change it to raise an error if all values for a column are empty.

Reply:
changed it back to assign default min/max values if all values for a key is empty.

Reply:
hmm, actually assigning default numbers implies there were values other than null in the column. I think in that case the min and max should be null as well

Reply:
that means I need to `return self` instead of raise error here.

https://github.com/kbase/collections/blob/main/src/common/product_models/columnar_attribs_common_models.py#L126-L127

are you ok with that?

Reply:
Yeah, but can you add a comment explaining that some columns may have all null values

Reply:
👍

Reply:
: